src/main/java/homework_6/map_problems/MapProblemsMutableGenerator.java

```diff
@@ -0,0 +1,34 @@
+package homework_6.map_problems;
+
+import java.awt.*;
+import java.util.Random;
+
+public class MapProblemsMutableGenerator extends MapProblemsGenerator {
+
+    public MapProblemsMutableGenerator(String clazzName, int clazzAge, Color clazzColor) {
+        super(clazzName, clazzAge, clazzColor);
+    }
+
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) return true;
+        if (o == null || getClass() != o.getClass()) return false;
+        MapProblemsMutableGenerator that = (MapProblemsMutableGenerator) o;
+        return clazzAge == that.clazzAge && clazzName.equals(that.clazzName) && clazzColor.equals(that.clazzColor);
+    }
+
+    @Override
+    public int hashCode() {
+        return new Random().nextInt(10);
+//        return new Random().nextInt();
+    }
```

Comment:
The problem with mutable keys is that you can't find the value if you change the key. In your case, it's just hashcode doesn't follow contracts. Even with properly-written hashcode mutable objects as keys can be lost in case of changing (and rehashing as the result)